clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1049,6 +1055,8 @@ scheduler_tmpl_t<RecordType, ReaderType>::get_initial_timestamps()
                 RecordType record = **input.reader;
                 if (record_type_is_timestamp(record, input.next_timestamp))
                     break;
+                if (record_type_is_instr(record))
```

Comment:
> Adds read-ahead to timestamps when replaying, to match ordinals when recording.

I was trying to figure out how issues in timestamp read-ahead can cause ordinal mismatches. Is L1058 related to it?

Reply:
I wanted to be able to compare both instr and record ordinals at start and context switch points in the input streams, and the read-ahead during record but not during replay made the input stream reported ordinals not match in tests.  (Would be better to have the records queued in the scheduler reflected in the input stream ordinals but that is not simple.)

Reply:
Updated the PR description on this.